### PR TITLE
Update sts configuration

### PR DESCRIPTION
**Pull request by MarcosDY**
Update sts configuration to allow set credentials

### CODE REVIEW

File: pkg/server/plugin/nodeattestor/aws/session.go [hunk 1]
```diff
@@ -35,22 +38,38 @@ func (cfg *SessionConfig) Validate(defaultAccessKeyID, defaultSecretAccessKey st
...
-func newAWSConfig(accessKeyID, secretAccessKey, region, assumeRoleArn string) aws.Config {
-	var credsProvider aws.CredentialsProvider
-	switch {
-	case assumeRoleArn != "":
-		stsConf := aws.Config{
-			Region: region,
+func newAWSConfig(accessKeyID, secretAccessKey, region, assumeRoleArn string) (aws.Config, error) {
+	ctx := context.Background()
+
+	var opts []func(*config.LoadOptions) error
```
Comment (rturner3) [inline pkg/server/plugin/nodeattestor/aws/session.go:44]: It's a little unfortunate that `config.LoadDefaultConfig()` accepts a `...func(*config.LoadOptions) error` rather than a `...config.LoadOptionsFunc`

File: pkg/server/plugin/nodeattestor/aws/session.go [hunk 2]
```diff
@@ -35,22 +38,38 @@ func (cfg *SessionConfig) Validate(defaultAccessKeyID, defaultSecretAccessKey st
...
+		)
 
-	return aws.Config{
-		Credentials: credsProvider,
-		Region:      region,
+		stsConf, err = config.LoadDefaultConfig(ctx, opts...)
+		if err != nil {
+			return aws.Config{}, err
+		}
 	}
```
Comment (rturner3) [inline pkg/server/plugin/nodeattestor/aws/session.go:73]: For readability, can we move this into a new helper method like `newAWSAssumeRoleConfig(region string, stsConf aws.Config)`? It gets a little hard to follow what options are being used where as currently written.

File: pkg/server/plugin/nodeattestor/aws/session.go (hunk 1 above)
Comment (MarcosDY) [reply pkg/server/plugin/nodeattestor/aws/session.go:44]: yeap

File: pkg/server/plugin/nodeattestor/aws/client.go [hunk 3]
```diff
@@ -20,49 +21,55 @@ type Client interface {
...
-	client, ok := cc.clients[region]
-	cc.mu.RUnlock()
-	if ok {
-		return client, nil
+	// Grab (or create) the cache for the region
+	r := cc.getCachedClient(cacheKey)
+
+	// Obtain the "lock" to the region cache
+	select {
+	case <-ctx.Done():
```
Comment (rturner3) [inline pkg/server/plugin/nodeattestor/aws/client.go:60]: Just confirming, is this change in the synchronization of concurrent map access driven by us trying to respect context cancellations during execution of this method?

**Review by rturner3 — APPROVED** [review]
